### PR TITLE
Feature/op 539 ondelete callback

### PR DESCRIPTION
**Pull request by tsognong**

### CODE REVIEW

File: src/components/accessories/patientTherapy/patientTherapyTable/consts.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+];
+export const label = {
+  startDate: "Start",
+  endDate: "End",
+  medicalId: "Medical",
+  qty: "Quantity",
+  freqInDay: "FID",
+  unitID: "unitID",
+  note: "Note",
+  freqInPeriod: "FIP",
```
Comment (axelpezzo) [inline src/components/accessories/patientTherapy/patientTherapyTable/consts.ts:13]: We should keep those strings inside a component in order to use the t() hook to get the Translations.

File: src/components/accessories/table/Table.tsx [hunk 2]
```diff
@@ -147,6 +164,17 @@ const Table: FunctionComponent<IProps> = ({
...
+      <ConfirmationDialog
+        isOpen={openDeleteConfirmation}
+        title={t("common.delete")}
+        info={t("common.deleteconfirmation")}
+        icon={warningIcon}
+        primaryButtonLabel="OK"
+        secondaryButtonLabel="Dismiss"
+        handlePrimaryButtonClick={handleDelete}
+        handleSecondaryButtonClick={() => setOpenDeleteConfirmation(false)}
+      />
```
Comment (axelpezzo) [inline src/components/accessories/table/Table.tsx:177]: I reviewed your work and I think that the component's logic is OK.

The only thing i've noticed is that: maybe in order to be more exaustive, we should add the content's title with the confirmation message eg: Are you sure to delete the #title_of_the_row ?

File: src/components/accessories/patientTherapy/patientTherapyTable/consts.ts (hunk 1 above)
Comment (tsognong) [reply src/components/accessories/patientTherapy/patientTherapyTable/consts.ts:13]: It is done in another branch. It does'nt concern the delete feature.

File: src/components/accessories/table/Table.tsx (hunk 2 above)
Comment (tsognong) [reply src/components/accessories/table/Table.tsx:177]: Ok.

File: src/components/accessories/dateField/DateField.tsx [hunk 3]
```diff
@@ -25,7 +26,13 @@ const DateField: FunctionComponent<IProps> = ({
...
 
   useEffect(() => {
-    setValue(fieldValue === "" ? null : new Date(+fieldValue));
+    if (fieldValue === "") {
+      setValue(null);
+    } else if (moment(+fieldValue).isValid()) {
+      setValue(moment(+fieldValue).toDate());
+    } else {
+      setValue(moment(fieldValue, "DD/MM/YYYY").toDate());
+    }
```
Comment (axelpezzo) [inline src/components/accessories/dateField/DateField.tsx:35]: See the comment on #165 MR

File: src/components/accessories/patientOPD/patientOPDTable/PatientOPDTable.tsx [hunk 4]
```diff
@@ -57,8 +81,11 @@ const PatientOPDTable: FunctionComponent<IOwnProps> = ({
     return results;
   };
 
-  const onDelete = () => {
-    console.log("delete");
+  const onDelete = (row: OpdDTO) => {
+    if (row.code) {
+      setDeletedObjCode(row.code + "");
+      dispatch(deleteOpd(row.code));
+    }
```
Comment (axelpezzo) [inline src/components/accessories/patientOPD/patientOPDTable/PatientOPDTable.tsx:88]: I would show an error if something went wrong

**Review by axelpezzo — CHANGES_REQUESTED** [review]
See the https://github.com/informatici/openhospital-ui/pull/163/files#r682414864

File: src/components/accessories/patientOPD/patientOPDTable/PatientOPDTable.tsx (hunk 4 above)
Comment (tsognong) [reply src/components/accessories/patientOPD/patientOPDTable/PatientOPDTable.tsx]: OK. I have removed the if statement to handle undefined code directly in action creator.

File: src/components/accessories/patientOPD/patientOPDTable/PatientOPDTable.tsx [hunk 5]
```diff
@@ -57,8 +81,9 @@ const PatientOPDTable: FunctionComponent<IOwnProps> = ({
     return results;
   };
 
-  const onDelete = () => {
-    console.log("delete");
+  const onDelete = (row: OpdDTO) => {
+    setDeletedObjCode(row.code + "");
```
Comment (axelpezzo) [inline src/components/accessories/patientOPD/patientOPDTable/PatientOPDTable.tsx:85]: `row.code ?? ""`

Comment (tsognong) [reply src/components/accessories/patientOPD/patientOPDTable/PatientOPDTable.tsx:85]: The objCode to delete is a string. You can look at the initialization. It is just used in translation message.

Comment (axelpezzo) [reply src/components/accessories/patientOPD/patientOPDTable/PatientOPDTable.tsx:85]: The code `row.code ?? ""` is the same as `row.code + ""` but more javascript compliant.
If the code is undefined then return "".

Comment (tsognong) [reply src/components/accessories/patientOPD/patientOPDTable/PatientOPDTable.tsx:85]: Right. Thanks

Comment (tsognong) [reply src/components/accessories/patientOPD/patientOPDTable/PatientOPDTable.tsx]: I applied your suggestions

**Review by axelpezzo — APPROVED** [review]

**Comment by axelpezzo** [comment]
@tsognong there are some conflicts with the develop branch.

**Comment by tsognong** [comment]
OK, I am resolving the conflicts

**Pull request merged**